internal/server/recon/recognize.go

```diff
@@ -116,7 +118,17 @@ func RecognizeEntities(
 				entities = append(entities, &pb.RecognizeEntitiesResponse_Entity{Dcid: id})
 			}
 			for span := range id2spans[entity.GetInId()] {
-				span2item[span] = &pb.RecognizeEntitiesResponse_Item{Span: span, Entities: entities}
+				if types, ok := store.RecogPlaceStore.ReconNameToRequiredTypes[entity.GetInId()]; ok {
+					// If the recognized name has required types that it can resolve for,
+					// add the type in front of the span so only spans that include the
+					// type will resolve.
+					for _, t := range types {
+						span_with_type := strings.ToLower(t) + " " + span
```

Comment:
I wonder if we want alternate names (or space delimited name) for `t`?  Like BiologicalSpecimen should really be "biological specimen", rather than "biologicalspecimen".  Or "Mesh Descriptor".

Have a TODO?

Reply:
added alternate names because you're right, i think the user will at least space delimit the type